extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -140,32 +139,46 @@ function inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
   opt_vendor) {
   const win = a4aElement.win;
   const ampDoc = a4aElement.getAmpDoc();
-  if (Object.keys(seenUrls).length == MAX_RTC_CALLOUTS) {
-    return logAndAddErrorResponse_(
-        promiseArray, RTC_ERROR_ENUM.MAX_CALLOUTS_EXCEEDED,
-        opt_vendor || url);
-  }
+  /**
+   * The time that it takes to substitute the macros into the URL can vary
+   * depending on what the url requires to be substituted, i.e. a long
+   * async call. Thus, however long the URL replacement took is treated as a
+   * time penalty.
+   */
+  const send = (url, timePenalty) => {
+    if (Object.keys(seenUrls).length == MAX_RTC_CALLOUTS) {
```

Comment:
These first two (max count & not secure)  can be done immediately without waiting for url expansion, right?

Reply:
Unfortunately no. We don't know what a given URL will be expanded to necessarily until after it is expanded, so we can't say for sure ahead of time. Same goes for secure url, because parseUrl (which is called as part of that) might break on an invalid unexpanded URL.

Reply:
ugh ok.  BTW, do you ever call buildErrorResponse_ without logging?

Reply:
yes